crates/sui-core/src/execution_cache.rs

```diff
@@ -146,9 +144,19 @@ pub fn build_execution_cache(
     store: &Arc<AuthorityStore>,
 ) -> ExecutionCacheTraitPointers {
     let execution_cache_metrics = Arc::new(ExecutionCacheMetrics::new(prometheus_registry));
-    ExecutionCacheTraitPointers::new(
-        ProxyCache::new(epoch_start_config, store.clone(), execution_cache_metrics).into(),
-    )
+
+    if matches!(
+        epoch_start_config.execution_cache_type(),
+        ExecutionCacheConfigType::WritebackCache
```

Comment:
would it be safer (future proof) to make this an exhuastive match?

Reply:
there definitely won't be any new cache types - as soon as I can i will strip all of this conditional stuff out.